test/gtest/ucp/test_ucp_ep_reconfig.cc

```diff
@@ -97,29 +99,35 @@ class test_ucp_ep_reconfig : public ucp_test {
         add_variant_with_value(variants, UCP_FEATURE_TAG, 0, "");
     }
 
-    void run(unsigned init_flags = UCP_EP_INIT_CREATE_AM_LANE,
-             bool bidirectional = false);
+    void run(bool bidirectional = false);
 
     void send_message(const ucp_test_base::entity &e1,
                       const ucp_test_base::entity &e2, const mem_buffer &sbuf,
                       const mem_buffer &rbuf, size_t msg_size,
-                      std::vector<void*> &reqs)
+                      std::vector<void*> &reqs, bool bidirectional)
     {
         const ucp_request_param_t param = {
             .op_attr_mask = UCP_OP_ATTR_FLAG_NO_IMM_CMPL
         };
 
-        void *sreq = ucp_tag_send_nbx(e1.ep(), sbuf.ptr(), msg_size, 0, &param);
-        void *rreq = ucp_tag_recv_nbx(e2.worker(), rbuf.ptr(), msg_size, 0, 0,
-                                      &param);
+        auto tag_send = bidirectional ? ucp_tag_send_sync_nbx :

```

Comment:
why bidirectional requires ucp_tag_send_sync_nbx?

Reply:
`ucp_tag_send_sync_nbx` triggers resolve_remote_id

Reply:
how is it related to bidirectional? why not needed for uni directional?

Reply:
because it is triggered from send operation